assets/prototype/domain/eventEditor/ticketsList/ticketCard/useOnDeleteTicket.js

```diff
@@ -0,0 +1,25 @@
+import useEntityMutator from '../../containers/mutations/useEntityMutator';
+import useTicketPrices from '../../containers/queries/useTicketPrices';
+
+const useOnDeleteTicket = ({ id }) => {
+	const { deleteEntity: deleteTicket } = useEntityMutator('Ticket');
+	const relatedPrices = useTicketPrices(id);
+	const { deleteEntity: deletePrice } = useEntityMutator('Price');
+
+	return () => {
+		const subscriptions = {
+			onCompleted: () => {
+				// The prices that are not default or tax prices.
+				const pricesToDelete = relatedPrices.filter(({ isDefault, isTax }) => !isDefault && !isTax);
+				const priceIdsToDelete = pricesToDelete.map(({ id }) => id);
+				priceIdsToDelete.forEach((id) => {
+					deletePrice({ id });
+				});
```

Comment:
could have done the deletions right within the map no?

ie:

```js
pricesToDelete.map(({ id }) => deletePrice({ id }));
```

or were you planning to use those IDs for something else?

Reply:
Yeah, my idea was to keep track of the mutations which might fail, so we can identify them with those IDs.